x-pack/plugins/upgrade_assistant/public/application/components/overview/logs_step/logs_step.tsx

```diff
@@ -0,0 +1,224 @@
+import { loadLogsCheckpoint } from '../../../lib/logs_checkpoint';
+import type { OverviewStepProps } from '../../types';
+import { useDeprecationLogging } from '../../es_deprecation_logs';
+
+const i18nTexts = {
+  logsStepTitle: i18n.translate('xpack.upgradeAssistant.overview.logsStep.title', {
+    defaultMessage: 'Address deprecation logs',
+  }),
+  logsStepDescription: i18n.translate('xpack.upgradeAssistant.overview.logsStep.description', {
+    defaultMessage:
+      'Review the Elasticsearch deprecation logs to make sure you are not using deprecated APIs.',
+  }),
+  viewLogsButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.viewLogsButtonLabel',
+    {
+      defaultMessage: 'View logs',
+    }
+  ),
+  enableLogsButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.enableLogsButtonLabel',
+    {
+      defaultMessage: 'Enable logs',
+    }
+  ),
+  logsCountDescription: (deprecationCount: number, checkpoint: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.overview.logsStep.countDescription"
+      defaultMessage="You have {deprecationCount, plural, =0 {no} other {{deprecationCount}}} deprecation {deprecationCount, plural, one {issue} other {issues}} since {checkpoint}."
+      values={{
+        deprecationCount,
+        checkpoint: (
+          <>
+            <FormattedDate value={checkpoint} year="numeric" month="long" day="2-digit" />{' '}
+            <FormattedTime value={checkpoint} timeZoneName="short" hour12={false} />
+          </>
+        ),
+      }}
+    />
+  ),
+  missingPrivilegesTitle: i18n.translate(
+    'xpack.upgradeAssistant.overview.logsStep.missingPrivilegesTitle',
+    {
+      defaultMessage: 'You require index privileges to analyze the deprecation logs',
+    }
+  ),
+  loadingError: i18n.translate('xpack.upgradeAssistant.overview.logsStep.loadingError', {
+    defaultMessage: 'An error occurred while retrieving the count of deprecation logs',
+  }),
+  retryButton: i18n.translate('xpack.upgradeAssistant.overview.logsStep.retryButton', {
+    defaultMessage: 'Try again',
+  }),
+};
+
+interface LogStepProps {
+  setIsComplete: (isComplete: boolean) => void;
+  hasPrivileges: boolean;
+  navigateToEsDeprecationLogs: () => void;
+}
+
+const LogStepDescription = () => (
+  <EuiText>
+    <p>{i18nTexts.logsStepDescription}</p>
+  </EuiText>
+);
+
+const LogsStep = ({ setIsComplete, hasPrivileges, navigateToEsDeprecationLogs }: LogStepProps) => {
+  const {
+    services: { api },
+  } = useAppContext();
+
+  const { isDeprecationLogIndexingEnabled } = useDeprecationLogging();
+
+  const checkpoint = loadLogsCheckpoint();
+
+  const {
+    data: logsCount,
+    error,
+    isLoading,
+    resendRequest,
+    isInitialRequest,
+  } = api.getDeprecationLogsCount(checkpoint);
+
+  useEffect(() => {
+    if (!isDeprecationLogIndexingEnabled) {
+      setIsComplete(false);
+    }
+
+    setIsComplete(logsCount?.count === 0);
```

Comment:
I love that we're bubbling this state up to the steps UI. If this step is marked as incomplete, does it block the user from upgrading? If so, then I'm concerned about users being blocked from upgrading in the event an Elastic-triggered log shows up and they have no way to fix it. Maybe I'm just being paranoid, but given the trouble we had with these logs originally I think we might want to err on the side of caution.

Reply:
Ah, good catch! I didn't realize we were blocking the upgrade on cloud, but now I see that we are. I will remove this.

Reply:
Actually, on further review, I see that we are calling the [status](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/routes/status.ts) route to determine whether or not the[ upgrade button is disabled](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/public/application/components/overview/upgrade_step/upgrade_step.tsx#L102). This takes into account any critical deprecations from ES or Kibana, and whether the system indices migration has completed. 

With that said, I think the current approach is OK to leave as-is.

Reply:
👍 Thanks for looking into it. Just wanted to make sure the actual button in UA that takes you to Cloud isn't disabled by the incomplete state of the step. But I believe that button is enabled/disabled based on the Status API, right?